### PR TITLE
fix(request): add one more condition to check doc is url or not

### PR DESCRIPTION
**Pull request by Kavan72**

### CODE REVIEW

**Comment by JoanFM** [comment]
Hey @Kavan72,

Could you please add a unittest that shows that the issue is indeed solved?

**Comment by Kavan72** [comment]
sure

**Comment by Kavan72** [comment]
Added unittest request generator

**Comment by nan-wang** [comment]
@Kavan72 Many thanks for your efforts! resolves #776

**Comment by JoanFM** [comment]
Related to: 
https://github.com/jina-ai/examples/issues/105

**Comment by JoanFM** [comment]
recheckcla

**Comment by JoanFM** [comment]
Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
#553 :bookmark: Commit Message Guideline for the First Time Contributor
jina-ai/jina | Jun 21st | Added by GitHub
CONTRIBUTING.md
jina-ai/jina | Added by GitHub

**Comment by Kavan72** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Kavan72** [comment]
@JoanFM done

**Comment by JoanFM** [comment]
recheckcla

File: tests/unit/test_index.py [hunk 1]
```diff
@@ -162,6 +162,17 @@ def test_index(self):
         with f:
             f.search(input_fn=random_docs(2), output_fn=get_result)
 
+    def test_index_request_generator(self):
```
Comment (JoanFM) [inline tests/unit/test_index.py:165]: what are we really testing here?

Comment (Kavan72) [reply tests/unit/test_index.py:165]: my bad, let me create new pull req cuz i forgot to create new branch in my fork

File: jina/helper.py [hunk 2]
```diff
@@ -597,13 +597,34 @@ def get_full_version() -> str:
 
 
 def is_url(text):
+    ul = '\u00a1-\uffff'  # Unicode letters range (must not be a raw string).
```
Comment (JoanFM) [inline jina/helper.py:600]: @nan-wang , shouldn't we have the `url_pat` regex compiled statically? outside this function? This compilation could slow a little bit the `request` generation right?

File: tests/unit/test_index.py (hunk 1 above)
Comment (JoanFM) [reply tests/unit/test_index.py]: you can change it in this PR if you need to change it.

But I fail to see what this test is actually doing.

Comment (Kavan72) [reply tests/unit/test_index.py]: i'm checking can we get req list even if doc is starting with url. in the old code it's throwing error on this part

File: tests/unit/test_request.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import unittest
+
+from jina.clients.python.request import _generate
+from tests import JinaTestCase
+
+
+class MyTestCase(JinaTestCase):
```
Comment (JoanFM) [inline tests/unit/test_request.py:7]: can u please add a more descriptive class name? we are trying to improve this.

File: tests/unit/test_request.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    def test_request_generate(self):
+
+        def random_lines(num_lines):
+            for j in range(num_lines):
+                yield "https://github.com 'i\'m dummy doc %d'" % j
+
+        req = _generate(random_lines(100))
+
+        self.assertEqual(len(list(req)), 1)
```
Comment (JoanFM) [inline tests/unit/test_request.py:17]: why does it expect it to just return 1 element?

Comment (JoanFM) [reply tests/unit/test_request.py:17]: Is it because of the `batch_size`? shouldn't it return 100 documents?

Comment (Kavan72) [reply tests/unit/test_request.py:17]: yes if we pass `batch_size=0` than it's return 1 req with all docs and if we pass `batch_size=10` than it's return 10 req with 10 docs each

Comment (JoanFM) [reply tests/unit/test_request.py:17]: so can you pass `batch_size=100`? I think this way seems a more explicit test

Comment (Kavan72) [reply tests/unit/test_request.py:17]: yep

Comment (Kavan72) [reply tests/unit/test_request.py:17]: done

Comment (JoanFM) [reply tests/unit/test_request.py:17]: but the assertion should not return 100 elements?

File: tests/unit/test_request.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import unittest
```
Comment (JoanFM) [inline tests/unit/test_request.py:1]: remove unittest

**Comment by JoanFM** [comment]
LGTM! Just some minor issues. We can handle moving the `url_pattern` regex outside the function scope in a different PR

File: tests/unit/test_request.py (hunk 4 above)
Comment (Kavan72) [reply tests/unit/test_request.py:17]: no no 
like if `batch_size=100` than it's return 1 req with 100 docs
if we pass `batch_size=100` than it's return 10 req with 10 docs each

Comment (JoanFM) [reply tests/unit/test_request.py:17]: ah, then can we `assert` that the request contains 100 documents?

File: tests/unit/test_request.py (hunk 5 above)
Comment (Kavan72) [reply tests/unit/test_request.py:1]: if i remove unittest than we cant call directly 
```
if __name__ == '__main__':
    unittest.main()```

Comment (JoanFM) [reply tests/unit/test_request.py:1]: remove that also, we are using `pytest` currently

File: tests/unit/test_request.py (hunk 4 above)
Comment (Kavan72) [reply tests/unit/test_request.py:15]: yes, like that
`req = _generate(data=random_lines(100), batch_size=100)`
len of req list is 1 but on that req we have 100 docs. so, i'm checking with len of req's list

File: tests/unit/test_request.py (hunk 5 above)
Comment (Kavan72) [reply tests/unit/test_request.py]: done

File: jina/helper.py (hunk 2 above)
Comment (hanxiao) [reply jina/helper.py:600]: yes, `url_pat` should be moved outside

**Comment by hanxiao** [comment]
Thanks for your contribution ❤️ 

Oh btw, we have a slack channel for warm-heart contributors like you, where we discuss code, PR, and future of Jina. Strongly recommend to join us

- If you haven't joined our Slack, [click here to join](https://join.slack.com/t/jina-ai/shared_invite/zt-dkl7x8p0-rVCv~3Fdc3~Dpwx7T7XG8w)
- 👨‍💻 👩‍💻 ➡️  https://jina-ai.slack.com/messages/contributors/

**Comment by Kavan72** [comment]
sure i'll join stack

**Comment by alexcg1** [comment]
@all-contributors please add @Kavan72  for code

**Pull request merged**